### ISSUE TITLE
Ignore patterns should be editable at folder creation time. 

### ISSUE THREAD

**Issue by calmh** [syncthing/syncthing#3608]
In order to prevent files from being scanned during initial scan.

**Comment by AudriusButkevicius** [comment]
Why does this matter?

**Comment by calmh** [comment]
Because the workflow to work around it otherwise is crap. Consider that you get the "share folder" prompt because another device want to share the photos folder with you. You have a photos folder to share, but there are some files in it that should not be shared. Today you'd have to create the folder, make sure it's not shared with another device, then set the ignore patterns and make sure a rescan happens, and then share the folder with the other device. 

That's apart from the generally work saving measure of excluding the large ISO files you don't want shared, at  creation time. A scan isn't cancelable so otherwise you're stuck waiting for the unnecessary scan to complete before the files can be ignored. And even then they'll be part of the index exchange...

**Comment by scienmind** [comment]
It also may be good idea to streamline "global ignores" here:
Right now, if you define `#include .stglobalignore` before the folder is actually in sync, it'll mark the folder as "Error", complaining that the file mentioned in `#include` is missing.
(not ideal, since you are going to sync "unwanted files" as well, just to get the `.globalignore` files)

Best case would be: 
1 - allow ignores editing as calmh mentioned upon, folder creation 
2 - if user entered `#include`s there - don't fail straight away, and try to fetch files that are mentioned in `#include`  lines first. Settle the full ignore list with inclusions. Cary on to the rest of the synchronizations.

**Comment by bigbear2nd** [comment]
> @calmh  And even then they'll be part of the index exchange...

I might add: forever. :smile: , See #863 

-- 
It would also be perfect to ask the user then, if he might want to take a "standard" ignored list into action. (At least it might be possible then for the wrappers)

**Comment by bugith** [comment]
Reading post and feature request for Pause all folders or devices, I idea hit my mind for this 3608:
We could either have a "Paused" check box in the dialog, or an additional button at the bottom : "_Save paused_" (if possible with a tooltip "_Useful e.g. to create ignore patterns before first sync_").

Whatever, we yet have to workarounds : 1st = don't populate the directory before setting stignore ; 2nd = Pause the intended devices before we start creating a folder that would need stignore.

**Comment by bugith** [comment]
Hi guys
Today with 0.14.27rc3 I had the opportunity to test creating .stignore on folder creation. When the "Ignore Patterns" button is grayed out OK when no path is given yet (good thing), and becomes available only when a path is given. 
The problem is that the file is not created on "Save" ignores (even if the directory yet exists). The whole path would be created at this moment, before the share is when .stfolder is created on second Save click.
Only the file is updated when it pre-exists.

**Comment by calmh** [comment]
@bugith That sounds like a bug. Please file it as a new issue.

**Comment by calmh** [comment]
Oh, and thank you for catching this in a pre-release. It's much appreciated, my curt reply about the process of things notwithstanding. :)

**Comment by bugith** [comment]
> done
